python_modules/dagster/dagster/core/definitions/time_window_partitions.py

```diff
@@ -117,6 +117,19 @@ def get_default_partition_mapping(self):
 
         return TimeWindowPartitionMapping()
 
+    @staticmethod
+    def get_partitions_from_serialized_data(
+        schedule_type: ScheduleType, start: float, timezone: str, fmt: str, end_offset: int
```

Comment:
Would it make sense for this to just accept a `ExternalTimeWindowPartitionsData`? Or for `ExternalTimeWindowPartitionsData` to have a method that returns a `TimeWindowPartitionsDefinition`?

Reply:
I'll add a method to `ExternalTimeWindowPartitionsData` that returns `TimeWindowPartitionsDefinition`. Only reason why this method didn't accept the `ExternalTimeWindowPartitionsData` was because of circular imports